python/TestHarness/testers/RunApp.py

```diff
@@ -235,18 +235,12 @@ def processResults(self, moose_dir, retcode, options, output):
         out_ok = self.checkOutputForLiteral(output, specs['expect_out'])
       else:
         out_ok = self.checkOutputForPattern(output, specs['expect_out'])
-      # Process out_ok
-      if (out_ok and retcode != 0):
-        reason = 'OUT FOUND BUT CRASH'
-      elif (not out_ok):
+      if (not out_ok):
         reason = 'NO EXPECTED OUT'
 
-    elif specs.isValid('absent_out'):
-      out_ok = self.checkOutputForPattern(output, specs['absent_out'])
-      # Process out_ok
-      if (not out_ok and retcode != 0):
-        reason = 'OUTPUT ABSENT BUT CRASH'
-      elif (out_ok):
+    if reason == '' and specs.isValid('absent_out'):
+      out_ok = not self.checkOutputForPattern(output, specs['absent_out'])
+      if (not out_ok):
```

Comment:
The double-negative on lines 242 and 243 is kind of confusing, but I think it's correct...